docs/index.bs

```diff
@@ -136,155 +90,193 @@ spec: rfc7230; urlPrefix: https://tools.ietf.org/html/rfc7230
-    <p>A <a href="#dfn-service-worker">service worker</a> has an associated <dfn id="dfn-skip-waiting-flag">skip waiting flag</dfn>. Unless stated otherwise it is unset.</p>
-    <p>A <a href="#dfn-service-worker">service worker</a> has an associated <dfn id="dfn-imported-scripts-updated-flag">imported scripts updated flag</dfn>. It is initially unset.</p>
-    <p>A <a href="#dfn-service-worker">service worker</a> has an associated <dfn id="dfn-set-of-event-types-to-handle">set of event types to handle</dfn> whose element type is an <a>event listener</a>'s event type. It is initially set to null.</p>
-    <p>A <a href="#dfn-service-worker">service worker</a> has an associated <dfn id="dfn-foreign-fetch-scopes">list of foreign fetch scopes</dfn> whose element type is a <a for="/">URL</a>. It is initially empty.</p>
-    <p>A <a href="#dfn-service-worker">service worker</a> has an associated <dfn id="dfn-foreign-fetch-origins">list of foreign fetch origins</dfn> whose element type is a <a for="/">URL</a>. It is initially empty.</p>
+    A <dfn export id="dfn-service-worker" for="">service worker</dfn> is a type of <a>web worker</a>. A <a for="/">service worker</a> executes in the registering <a for="/">service worker client</a>'s <a for="/">origin</a>.
+
+    A <a for="/">service worker</a> has an associated <dfn export id="dfn-state">state</dfn>, which is one of *parsed*, *installing*, *installed*, *activating*, *activated*, and *redundant*. It is initially *parsed*.
+
+    A <a for="/">service worker</a> has an associated <dfn export id="dfn-script-url">script url</dfn> (a <a for="/">URL</a>).
+
+    A <a for="/">service worker</a> has an associated <dfn export id="dfn-type">type</dfn> which is either "<code>classic</code>" or "<code>module</code>". Unless stated otherwise, it is "<code>classic</code>".
+
+    A <a for="/">service worker</a> has an associated <dfn export id="dfn-containing-service-worker-registration" lt="registration|containing service worker registration">containing service worker registration</dfn> (a <a for="/">service worker registration</a>), which contains itself.
+
+    A <a for="/">service worker</a> has an associated <dfn export id="dfn-service-worker-id">id</dfn> (an opaque string), which uniquely identifies itself during the lifetime of its <a>containing service worker registration</a>.
+
+    A <a for="/">service worker</a> has an associated <dfn export id="dfn-service-worker-global-object" for="service worker">global object</dfn> (a {{ServiceWorkerGlobalScope}} object or null).
+    
+    A <a for="/">service worker</a> is dispatched a set of <dfn export id="dfn-lifecycle-events">lifecycle events</dfn>, <a event for="ServiceWorkerGlobalScope">install</a> and <a event for="ServiceWorkerGlobalScope">activate</a>, and <dfn export id="dfn-functional-events">functional events</dfn> including <a event for="ServiceWorkerGlobalScope">fetch</a>.
```

Comment:
Let's use `{{install!!event}}`. I think it'd not be ambiguous without the event target within the spec.